lib/srv/reexec.go

```diff
@@ -114,6 +106,21 @@ type UaccMetadata struct {
 	WtmpPath string `json:"wtmp_path,omitempty"`
 }
 
+// PAMConfig stores PAM specific configuration passed to the child.
+type PAMConfig struct {
+	// Enabled indicates if PAM support was requested by the node.
+	Enabled bool `json:"enabled"`
```

Comment:
This could be done away with if you used the `PAMConfig` value availability (e.g. whether the pointer is not `nil`)  as the indicator whether PAM has been enabled - unless there is requirement to pass the rest of the attributes even if `Enabled == false`?